src/test/scala/com/fulcrumgenomics/bam/SamRecordClipperTest.scala

```diff
@@ -758,4 +758,18 @@ class SamRecordClipperTest extends UnitSpec with OptionValues {
     mate.start shouldBe 1
     mate.cigar.toString shouldBe "40M20I40M"
   }
+
+  it should "clip FR reads when the F read doesn't align past the R read but contains soft-clipped bases that extend past" in {
+    val builder = new SamBuilder(readLength=50, sort=Some(Queryname))
+    val clipper = new ClipBam(input=dummyBam, output=dummyBam, ref=ref, clipBasesPastMate=true)
+    val Seq(r1, r2) = builder.addPair(start1=20, start2=20, strand1=Minus, strand2=Plus, bases1="A"*10+"CGT"+"C"*37, bases2="CGT"+"C"*35+"A"*12, cigar1="10S40M", cigar2="38M12S")
```

Comment:
can you use `pair` helper method instead of builder/addpair?

Reply:
also this test could test the output of `SamRecordClipper.clipExtendingPastMateEnds` (thats what `it` is in this context)

Reply:
I got git-lazy.  I made fixes in the original clone and manually brought them over to a fork but didn't pay close enough attention and pasted the test at the end of the wrong file.  Fix incoming.

Reply:
I do think its a good idea to add tests to `SamRecordClipperTest` that cover these cases since it is closer to the code change you are submitting